src/services/outliningElementsCollector.ts

```diff
@@ -3,12 +3,18 @@ namespace ts.OutliningElementsCollector {
     export function collectElements(sourceFile: SourceFile, cancellationToken: CancellationToken): OutliningSpan[] {
         const elements: OutliningSpan[] = [];
         const collapseText = "...";
+        let depth = 0;
+        const maxDepth = 20;
+
+        walk(sourceFile);
+        return elements;
 
-        function addOutliningSpan(hintSpanNode: Node, startElement: Node, endElement: Node, autoCollapse: boolean) {
+        // If useFullStart is true, then the collapsing span includes leading whitespace, including linebreaks.
+        function addOutliningSpan(hintSpanNode: Node, startElement: Node, endElement: Node, autoCollapse: boolean, useFullStart: boolean) {
             if (hintSpanNode && startElement && endElement) {
                 const span: OutliningSpan = {
-                    textSpan: createTextSpanFromBounds(startElement.pos, endElement.end),
-                    hintSpan: createTextSpanFromNode(hintSpanNode, sourceFile),
+                    textSpan: createTextSpanFromBounds(useFullStart ? startElement.pos : startElement.getStart(), endElement.end),

```

Comment:
Consider also calling `createTextSpanFromNode` and adding a boolean parameter to use full start that defaults to false.

Regardless of whether you do it inline here or by modifying the helper, use `startElement.getFullStart(...)`  to get the full start.